src/controllers/activitiesController.js

```diff
@@ -32,33 +32,32 @@ const saveActivity = async (req, res) => {
 // @access  public
 const updateActivity = async (req, res) => {
   const { _id } = req.params;
-  const { tally } = req.body;
-  if (!tally || isNaN(tally)) {
+  const { name, description, category } = req.body;
+  if (!name) {
     res.status(400);
-    throw new Error("An integer vote tally value must be provided!");
-  } else if (tally !== 1 && tally !== -1) {
+    throw new Error("New activity name must be provided!");
+  } else if (!category) {
     res.status(400);
-    throw new Error(
-      "Provide 1 to increase, and -1 to decrease the vote tally!"
-    );
+    throw new Error("New activity category must be provided!");
   }
```

Comment:
What happens when we try to change the category to an invalid option?

Reply:
The `category` filed of the Activity model is set to an enum. Like so:
category: {
      type: String,
      enum: {
        values: [
          "relaxing",
          "creative",
          "social",
          "educational",
          "holiday",
          "cultural",
        ],
        message:
          "Invalid category type! Choose from ['relaxing', 'creative', 'social', 'educational', 'holiday', 'cultural']",
      },
      required: [true, "please provide a valid category!"],
      
And if not a valid (preset) category is entered, the user will receive the `message`.

Reply:
Okay, I see -- your `errorHandler` function is actually doing the logic of converting the validation error to the proper response. Nice! 👍